features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Create/GraphQL/Steps/ProvideGraphQL.jsx

```diff
@@ -0,0 +1,277 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { Component } from 'react';
+import {
+    withStyles,
+    FormHelperText,
+} from '@material-ui/core';
+import ErrorOutline from '@material-ui/icons/ErrorOutline';
+import Typography from '@material-ui/core/Typography';
+import PropTypes from 'prop-types';
+import Progress from 'AppComponents/Shared/Progress';
+import { FormattedMessage } from 'react-intl';
+import Dropzone from 'react-dropzone';
+import classNames from 'classnames';
+import Backup from '@material-ui/icons/Backup';
+import Icon from '@material-ui/core/Icon';
+import API from 'AppData/api';
+
+const styles = theme => ({
+    radioWrapper: {
+        display: 'flex',
+        flexDirection: 'row',
+    },
+    dropZoneInside: {},
+    dropZone: {
+        width: '100%',
+        color: theme.palette.grey[500],
+        border: 'dashed 1px ' + theme.palette.grey[500],
+        background: theme.palette.grey[100],
+        padding: theme.spacing.unit * 4,
+        textAlign: 'center',
+        cursor: 'pointer',
+    },
+    dropZoneIcon: {
+        color: theme.palette.grey[500],
+        width: 100,
+        height: 100,
+    },
+    fileIcon: {
+        width: 100,
+        height: 100,
+        size: 32,
+    },
+    dropZoneError: {
+        color: theme.palette.error.main,
+    },
+    dropZoneErrorBox: {
+        border: 'dashed 1px ' + theme.palette.error.main,
+    },
+    errorMessage: {
+        color: theme.palette.error.main,
+    },
+    errorIcon: {
+        color: theme.palette.error.main,
+        marginRight: theme.spacing.unit * 2,
+    },
+    fileNameWrapper: {
+        flexDirection: 'row',
+        alignItems: 'center',
+        '& div': {
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+        },
+    },
+    largeIcon: {
+        fontSize: 60,
+    },
+    FormControl: {
+        padding: 0,
+        width: '100%',
+        marginTop: 0,
+    },
+    errorMessageWrapper: {
+        display: 'flex',
+        alignItems: 'center',
+    },
+    urlWrapper: {
+        display: 'flex',
+        alignItems: 'center',
+    },
+    button: {
+        whiteSpace: 'nowrap',
+    },
+});
+
+/**
+ *
+ * @class ProvideGraphQL
+ * @extends {Component}
+ */
+class ProvideGraphQL extends Component {
+    /**
+     * Creates an instance of ProvideGraphQL.
+     * @param {any} props @inheritDoc
+     * @memberof ProvideGraphQL
+     */
+    constructor(props) {
+        super(props);
+        const { file } = props;
+        this.state = {
+            file,
+            isValid: null,
+            errorMessage: '',
+            loading: false,
+        };
+        this.validateGraphQL = this.validateGraphQL.bind(this);
+    }
+
+    /**
+     * @inheritDoc
+     * @param {any} nextProps New props received
+     * @memberof ProvideGraphQL
+     */
+    componentWillReceiveProps(nextProps) {
```

Comment:
this life cycle hook is not recommend to use now [1]. since you are doing an side effect we can use componentDidUpdate or can we use componentDidMount? 

[1] - [https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)